packages/embark-ipfs/src/index.js

```diff
@@ -72,8 +67,35 @@ class IPFS {
 
       upload_ipfs.deploy(readyCb);
     });
+  }
+
+  async setupIpfsApi() {
+    this.events.request("runcode:whitelist", 'ipfs-api', () => {});
+    this.events.on("storage:started", this.registerIpfsObject.bind(this));
+    this.registerIpfsObject();
+    this.registerIpfsHelp();
+  }
+
+  async registerIpfsObject() {
+    const {host, port} = this.config;
+    let ipfs = IpfsApi(host, port);
+    await this.events.request2("runcode:register", "ipfs", ipfs);
+  }
+
+  async registerIpfsHelp() {
+    await this.events.request2('console:register:helpCmd', {
+      cmdName: "ipfs",
+      cmdHelp: __("instantiated js-ipfs object configured to the current environment (available if ipfs is enabled)")
+    });
+  }
 
-    this.registerEmbarkJSStorage();
+  setupEmbarkJS() {
+    this.events.request("embarkjs:plugin:register", 'storage', 'ipfs', 'embarkjs-ipfs');
+    this.events.request("embarkjs:console:register", 'storage', 'ipfs', 'embarkjs-ipfs');
```

Comment:
Shouldn't this be an `async` method with:
```suggestion
    await this.events.request2("embarkjs:console:register", 'storage', 'ipfs', 'embarkjs-ipfs');
```